### ISSUE TITLE
pupnp completely broken on actual Windows 10 PC

### ISSUE THREAD

**Issue by spl1inter** [msys2/MINGW-packages#6090]
Has anyone here _actually_ been able to browse content in the MINGW release of VLC, using "Universal Plug 'n' Play"?
 The console will give you a generic error like:
```
VLC media player - 3.0.8 Vetinari
Copyright © 1996-2019 the VideoLAN team

Warning: if you cannot access the GUI anymore, open a command-line window, go to the directory where you installed VLC and run "vlc -I qt"

Remote control interface initialized. Type `help' for help.
[0000018686e40940] upnp services discovery: Initializing libupnp on 'default' interface
[0000018686e40940] upnp services discovery error: Initialization failed: UPNP_E_INIT_FAILED
[0000018686e40940] main services discovery error: no suitable services discovery module
```

So if you know that pupnp actually provides this service, you can go back to the source and run it's tests for MinGW (test_init.exe):
```
UPNP_VERSION_STRING = "1.6.25"
UPNP_VERSION_MAJOR  = 1
UPNP_VERSION_MINOR  = 6
UPNP_VERSION_PATCH  = 25
UPNP_VERSION        = 10625

UPNP_HAVE_DEBUG         = yes
UPNP_HAVE_CLIENT        = yes
UPNP_HAVE_DEVICE        = yes
UPNP_HAVE_WEBSERVER     = yes
UPNP_HAVE_TOOLS         = yes

Initializing UPnP ...
** ERROR UpnpInit(): -117 UPNP_E_INIT_FAILED
```

Again an extremely helpful error, -117 basically is pupnp code for "failed for an unknown reason"

Apparently mingw-w64-pupnp package doesn't actually work at all. Please try it and let me know if you can actually get it working on a real, reasonably modern PC system.

**Comment by lazka** [comment]
It doesn't even build right now here:

```
In file included from ../../pupnp-release-1.6.25/upnp/src/inc/uri.h:57,
                 from ../../pupnp-release-1.6.25/upnp/src/inc/httpparser.h:42,
                 from ../../pupnp-release-1.6.25/upnp/src/ssdp/ssdp_ctrlpt.c:48:
../../pupnp-release-1.6.25/upnp/src/inc/inet_pton.h:34:20: error: conflicting types for 'inet_ntop'
   34 | extern const char *inet_ntop(int af, const void *src, char *dst,
      |                    ^~~~~~~~~
```

but, it looks like upstream got active again: https://github.com/pupnp/pupnp

and there are many new releases, so we should probably look into updating to those.

**Comment by spl1inter** [comment]
There appear to be several issues here.

> 
> 
> It doesn't even build right now here:
> 
> ```
> In file included from ../../pupnp-release-1.6.25/upnp/src/inc/uri.h:57,
>                  from ../../pupnp-release-1.6.25/upnp/src/inc/httpparser.h:42,
>                  from ../../pupnp-release-1.6.25/upnp/src/ssdp/ssdp_ctrlpt.c:48:
> ../../pupnp-release-1.6.25/upnp/src/inc/inet_pton.h:34:20: error: conflicting types for 'inet_ntop'
>    34 | extern const char *inet_ntop(int af, const void *src, char *dst,
>       |                    ^~~~~~~~~
> ```

This is probably because pupnp Windows support is out-of-date, even in the latest release. The only compiler they directly support is from Windows 7, and ntop/pton were introduced with SDK version 8.1. I was able to safely #ifdef those out of inet_pton.c and friends.


> but, it looks like upstream got active again: https://github.com/pupnp/pupnp
> 
> and there are many new releases, so we should probably look into updating to those.

Thanks for saying this. Now it works. TBH I thought there was some special reason we were still using the 1.6.x branch.  Things I did:

- Applied our MinGW x64 patch to current upstream release
- ifdef'd the aforementioned inet_pton definitions (only needed for very old MSVC versions)
- configured using **--disable-ipv6**, otherwise my library seemed to have BIND issues (this has always been a back-and-forth situation with pupnp afaik)

TL;DR updating to latest upstream works, although Windows support for this should be noted as very outdated and likely in need of broader review.

I'll leave closing this ticket up to you (we have progress but I wouldn't consider the issue totally closed just yet)

**Comment by lazka** [comment]
PR welcome :)

**Comment by revelator** [comment]
just need to set _WIN32_WINNT to 0x0501 the mingw-w64 api defaults to 0x0600 now and this package uses xp compatibility where inet_ntop etc. does not exist. Or you could comment the compatibility functions in source by wrapping them in guards for xp compatibility.

**Comment by stahta01** [comment]
I am trying to update the pupnp package right now. Tim S.

**Comment by stahta01** [comment]
@spl1inter this can now be closed, right?

Tim S.

**Comment by MehdiChinoune** [comment]
package was updated.
